File: src/android/Message.java
package com.gae.scaffolder.plugin;
import java.util.Date;
public class Message {
	private String id;
	private String title;
	private String body;
	private String senderId;
	
	private String thumbnail_url;
	
	private String thumbnail_hash;
	private String senderName;
	private Date arrivalTime;
	public Message(String id , String title , String body , String senderId,String senderName,String thumbnail_url,String tumbnail_hash,String arrivalTime)
	{ 
	this.id=id;
	this.title=title;
	this.body=body;
	this.senderId=senderId;
	this.senderName =senderName;
	this.thumbnail_url = thumbnail_url;
	
	this.thumbnail_hash = thumbnail_hash;
	if(arrivalTime==null)
	{
		this.arrivalTime=new Date();
		
	} else {
		this.arrivalTime=new Date(Long.parseLong(arrivalTime));
		
	}
	
	}
	
	public String getId(){return this.id;}
	public String getTitle(){return this.title;}
	public String getBody(){return this.body;}
	public String getSenderId(){return this.senderId;}
	public Long getArrivalTime(){return this.arrivalTime.getTime();}
	public String getSenderName(){return this.senderName;};

	public String getThumbnailHash(){return this.thumbnail_hash;}
	public String getThumbnailUrl(){return this.thumbnail_url;}
	
}

File: src/ios/Message.m
#import "Message.h"
@implementation Message


-(id)init
{
   self = [super init];
   
}

-(id) initWithDict :(NSDictionary *) dict withDate:(NSDate *)date
{
self=[super init];
if(self)
{
self.id=dict[@"id"];
self.title=dict[@"title"];
self.body=dict[@"body"];
self.senderId=dict[@"senderId"];
self.thumbnail_url=dict[@"thumbnail_url"];
self.thumbnail_hash=dict[@"thumbnail_hash"];
self.senderName=dict[@"senderName"];
self.arrivalTime = date;
if(self.arrivalTime)
{
self.arrivalTime=[[NSDate alloc] init];
}

}
return self;
}
-(NSDictionary *) getDict
{
 NSDictionary *dict = @{@"id": self.id,@"title":self.title,@"body":self.body,@"senderId":self.senderId
 ,@"thumbnail_url":self.thumbnail_url,@"thumbnail_hash":self.thumbnail_hash,@"senderName":self.senderName,@"arrivalTime":[[NSNumber alloc] initWithDouble:[self.arrivalTime timeIntervalSince1970]]};
   return dict;
}

@end
